library/src/common/rocsparse_trm_info.cpp

```diff
@@ -56,15 +56,22 @@ void rocsparse::trm_info_t::set_row_map(void* const value)
 {
     this->row_map = value;
 }
+
```

Comment:
`set_row_map`, `set_diag_ind`, `set_transposed_perm`, `set_transposed_row_ptr`, and `set_transposed_col_ind` are no longer used. I would advocate getting rid of any of the set methods that involve setting pointers to memory we allocate with hipMallocAsync since this is dangerous and the new way is to use the `get_ref_xxx` routines in the `hipMallocAsync`. Keeping the set methods related to setting already allocated memory is fine. Basically I think the row map, diag, and transpose related allocated memory should never be set but instead should use your new `get_ref_xxx` methods.